[UBS] - #4939-convert-delete-endpoint-to-patch

Markiro1 opened the pull request:
1) add new ErrorMessage for check deactivate courier
2) add root for endpoint deactivateCourier() method in SecurityConfig
3) rename endpoint in SuperAdminController
4) change return type deactivateCourier() from void to CourierDto 
5) implement deactivateCourier() method in SuperAdminServiceImpl
6) add test for deactivateCourier() method

Markiro1 committed:
1) added new ErrorMessage for check deactivate courier
2) added root for endpoint deactivateCourier() method in
SecurityConfig
3) rename endpoint in SuperAdminController
4) change return type deactivateCourier() from void to CourierDto
5) implemented deactivateCourier() method in SuperAdminServiceImpl
6) added test for deactivateCourier() method

Markiro1 committed:
1) added 2 test

Markiro1 committed:
1) change DELETE status to DEACTIVATED in CourierStatus enum
2) change the status of the courier to DEACTIVATED in SuperAdminServiceImpl, DeactivateChosenEntityRepository and in tests

RomaMocherniuk commented on core/src/test/java/greencity/controller/SuperAdminControllerTest.java:
```diff
@@ -23,6 +23,7 @@
 import greencity.exceptions.tariff.TariffAlreadyExistsException;
 import greencity.filters.TariffsInfoFilterCriteria;
 import greencity.service.SuperAdminService;
+import liquibase.pro.packaged.E;
```
unused import

RomaMocherniuk commented on core/src/test/java/greencity/controller/SuperAdminControllerTest.java:
```diff
@@ -416,6 +417,11 @@ void deactivateTariffTest() {
 
+    @Test
+    void deactivateCourier() throws Exception {
+        mockMvc.perform(patch(ubsLink + "/deactivateCourier/{id}", 1L)).andExpect(status().isOk());
```
please verify that `superAdminService.deactivateCourier(id)` was called with the expected `id`

RomaMocherniuk commented on service/src/test/java/greencity/service/ubs/SuperAdminServiceImplTest.java:
```diff
@@ -487,25 +487,34 @@ void includeBagException2Test() {
+        Courier courier = getCourier();
+        when(courierRepository.findById(anyLong())).thenReturn(Optional.of(courier));
+        courier.setCourierStatus(CourierStatus.DEACTIVATED);
+        assertThrows(BadRequestException.class, () -> superAdminService.deactivateCourier(anyLong()));
```
please add `verify`

RomaMocherniuk commented on service-api/src/main/java/greencity/constant/ErrorMessage.java:
```diff
@@ -67,6 +67,7 @@ public final class ErrorMessage {
     public static final String LOCATION_STATUS_IS_ALREADY_EXIST =
         "Current location already has status that's you wanna chose";
     public static final String COURIER_IS_NOT_FOUND_BY_ID = "Couldn't found courier by id: ";
+    public static final String CANNOT_DEACTIVATE_COURIER = "courier is already deactivated with id: ";
```
Should the message start with a capital letter?

Markiro1 committed:
1) change deactivateCourierThrowBadRequestException() test

Markiro1 committed:
1) delete unused import
2) added verify() for deactivateCourier() test in SuperAdminControllerTest
3) change first letter to capital in ErrorMessage

Markiro1 committed:
1) added SUPER_ADMIN_LINK

Markiro1 committed:
1) change SUPER_ADMIN_LINK

Markiro1 committed:
1) change SUPER_ADMIN_LINK
